src/main/java/com/microsoft/graph/httpcore/GraphTelemetryHandler.java

```diff
@@ -0,0 +1,95 @@
+package com.microsoft.graph.httpcore;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+
+import javax.annotation.Nonnull;
+
+import com.microsoft.graph.CoreConstants;
+import com.microsoft.graph.httpcore.middlewareoption.GraphClientOptions;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * Middleware responsible for adding telemetry information on SDK usage
+ * Note: the telemetry only collects anonymous information on SDK version and usage. No personal information is collected.
+ */
+public class GraphTelemetryHandler implements Interceptor{
+
+    private GraphClientOptions mGraphClientOptions;
+
+    public GraphTelemetryHandler(){
+        this.mGraphClientOptions = new GraphClientOptions();
+    }
+
+    @SuppressFBWarnings
+    //graphClientOptions exposes strings which are naturally immutable
+    public GraphTelemetryHandler(@Nonnull final GraphClientOptions graphClientOptions){
+        this.mGraphClientOptions = graphClientOptions;
+    }
+
+    @Override
+    @Nonnull
+    public Response intercept(@Nonnull final Chain chain) throws IOException {
+        final Request request = chain.request();
+        final Request.Builder telemetryAddedBuilder = request.newBuilder();
+
+        FeatureTracker featureTracker = request.tag(FeatureTracker.class);
+        if(featureTracker == null) {
+            featureTracker = new FeatureTracker();
+        }
+
+        //This assumes a call to graph will always include v1.0 or beta in the url
+        final String graphEndpoint = request.url().toString().contains("/v1.0/") ? "-v1.0" : "-beta";
```

Comment:
you can effectively use the v1 service lib to call beta endpoints and vice versa by changing the base url.
The reason why we're including the information in the SDK version header is to check which lib they are using, otherwise we'd not include is at all an use the request path to compute our telemetry, this information needs to come from the service lib.